server/models/availability.js

```diff
@@ -0,0 +1,54 @@
+const mongoose = require("mongoose");
+
+
+const dayInformation={
+    isAvaliable:{
+        type:Boolean,
+        default:'true'
+    },
+    startDate:{
+        type:Number,
+        min:0,
+        max:24,
+        required:isAvaliableForDay
+    },
+    endDate:{
+        type:Number,
+        min:0,
+        max:24,
+        required:isAvaliableForDay,
+        validate: [dateValidator, 'Start Date must be less than End Date']
+    }
```

Comment:
These look great! I just want to clarify our plan for handling times that are not on the hour (e.g. 17:50), could you share your thoughts?

Reply:
Yeah, that part makes a lot of sense 👍🏼  But how do we represent a time that's not an integer, that has decimal points? For example, what if `end` is `14.35`? Does this represent `14:35` on a clock? Do we intend to do some math to convert a decimal number into a representation of time instead (e.g. the `0.25` in `14.25` means `0.25 / 100 * 60 = 15 minutes, therefore it represents `14:15`)?

On the other hand, what value do we store if the user enters 14:30 as their start time? `14.3` or `14.5`? Or is it something else?

Reply:
Thanks for the detailed explanation. Yes, the logic is totally correct. The start time and end time can be stored in a date format but in the availability schema the date is not required only time is required so currently, I am planning to use a string to store the time.